components/raftstore/src/store/fsm/peer.rs

```diff
@@ -752,6 +752,9 @@ where
             }
             self.fsm.batch_req_builder.request = Some(cmd);
         }
+        // Update the state whether the peer is pending on applying raft
+        // logs if necesssary.
+        self.on_check_peer_complete_apply_logs();
```

Comment:
Maybe we only need this check for the first 5 minutes after restart?

Reply:
FYI. The state will be ignored, whether this peer is still under pending state for applying raft logs, after restart 5 minutes later. And the check is nearly no extra cost, as the `pending_on_apply` will be checked in advance.